### ISSUE TITLE
`$run(swagger = fn)` does not work for RSConnect

### ISSUE THREAD

**Issue by schloerke** [rstudio/plumber#468]
RSC does not want to call `$run` with special parameters... such as a swagger function.

It's also weird to do this for `entrypoint.R` as we do not have a way to define swagger info in `plumber.R`.

-----------------------

I propose we have a new plumber tag to define a swagger function that should take in `pr` and `spec`.  This follows the natural plumber.R flow.

Example:

```r
#* @get-swagger
function(pr, spec) {
  spec$info$title <- Sys.time()
  spec
}
```

Options for plumber tag...
* `#* @get-swagger`
* `#* @get-openapi`
* `#* @swagger`
* `#* @openapi`
* ...?

----------------------

This should also be paired with a programmatic handler function... 

* `pr$handle_swagger(fn)`
* `pr$handle_openapi(fn)`
* ...?

This'll get weird having mounted routers with different customizations.  I'm thinking the root router will be the only one to be called to avoid confusion / definition collisions.  

The `handle_swagger` function will be called from the root router who has all of the endpoints recursively added from the mounted routers and itself.

------------------

cc @blairj09 @aronatkins

**Comment by blairj09** [comment]
I prefer either `#* @swagger` or `#* @openapi`. Not sure if it makes sense to transition away from the Swagger naming convention, but if we're going to do it, now is as good a time as any.

**Comment by schloerke** [comment]
I'm torn.  

Context for the name war... Swagger (old plumber implementation) turned into OpenAPI2 which has been updated to OpenAPI3.  Plumber currently implements OpenAPI3 spec and displays it in SwaggerUI.  

I don't know which name to pick going forward.  OpenAPI makes sense.... and we'll just say it's displayed using Swagger.

**Comment by blairj09** [comment]
Yeah - it's a nuanced decision since OpenAPI refers to the specification and Swagger now refers to tools used to work with that specification. Since we would be defining the specification, I think it makes sense to use `#* @openapi`.

**Comment by schloerke** [comment]
Drastically changed (and fixed) in v1.0.0. Closing
